### PR TITLE
[adapter] Use Move's transactional test framework

### PR DESCRIPTION
**Pull request by tnowacki**
- Move has a transactional test framework designed for invoking the adapter in a series of transactions specified in a file
- Added a Sui instantiation of this
- Used it for some adapter tests (more to come)

### CODE REVIEW

**Review by sblackshear — APPROVED** [review]
Expected value tests FTW!

File: sui_programmability/adapter/transactional-tests/tests/publish/init_public.move [hunk 1]
```diff
@@ -1,6 +1,11 @@
+//# init --addresses Test=0x0
```
Comment (sblackshear) [inline sui_programmability/adapter/transactional-tests/tests/publish/init_public.move:1]: Does this have to go before the copyright? Seems a bit awkward + might make life difficult for the copyright linter

**Comment by lxfind** [comment]
Where can I find the list of commands I could use?

File: sui_programmability/adapter/transactional-tests/tests/publish/init_public.move (hunk 1 above)
Comment (tnowacki) [reply sui_programmability/adapter/transactional-tests/tests/publish/init_public.move:1]: There is a longstanding bug/todo that the file has to begin with a command. I should fix that

**Comment by tnowacki** [comment]
> Where can I find the list of commands I could use?

The test system uses clap, so you will get a command-line style help prompt on any invalid command. You can also do `--help` once you find a command

File: sui_programmability/adapter/transactional-tests/tests/publish/init_public.move (hunk 1 above)
Comment (tnowacki) [reply sui_programmability/adapter/transactional-tests/tests/publish/init_public.move:1]: https://github.com/move-language/move/pull/92

**Pull request merged**